spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/MilliSpan.java

```diff
@@ -66,9 +66,10 @@ public MilliSpan(long begin, long end, String name, String traceId, List<String>
 	private MilliSpan() {
 		this.begin = 0;
 		this.name = null;
-		this.traceId = null;
+		this.traceId = 0;
 		this.spanId = null;
 		this.processId = null;
+		this.parents = new ArrayList<>();
```

Comment:
oh.. ok actually looking at fields, looks like this object is somewhat mutable. I think this is a separate issue.